### PR TITLE
Add options for compiling without each of the libraries

### PR DESCRIPTION
**Pull request by piponazo**
After the discussion we had in the PR #23, I have created this new one with the implementation in which we agreed. Now we will have one option per library indicating if the library will be compiled or not (all of them are defaulted to True, except **python**).

### CODE REVIEW

**Comment by lasote** [comment]
Looks great! just the default shared detail.

File: conanfile.py [hunk 1]
```diff
@@ -3,15 +3,78 @@
...
+        "signals": [True, False],
+        "system": [True, False],
+        "test": [True, False],
+        "thread": [True, False],
+        "timer": [True, False],
+        "type_erasure": [True, False],
+        "wave": [True, False]
+    }
+
+    default_options = "shared=True", \
```
Comment (piponazo) [inline conanfile.py]: Ups! I was repeating too fast the last operation performed with Vim :stuck_out_tongue_closed_eyes: . I will fix it.

**Comment by lasote** [comment]
I checked if the package ids are kept with this change but it doesn't. I forgot again to tell you that the new variables have to be defaulted to False to keep the old shas. Sorry for all these changes, my fault. 

The option names should be "without_xxx" defaulted to False (so by default they are activated), except the old python option, that should be kept. I know that the python option name won't be consistent with the new variables but it was the exception, it was the only module deactivated by default.

I understand if you are tired to change again this recipe. I can change it for you.

Thanks so much
Luis

**Comment by piponazo** [comment]
No problem at all, I think that you told me the last time but since I had to work on other things I forgot about that. Please, feel free to change it if I do not do it in the next days ;)

**Comment by lasote** [comment]
looks great now! Thank you so much.

**Pull request merged**